hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -2130,15 +2124,18 @@ private void doRun() throws InterruptedException {
         }
 
         boolean processTask(Runnable r) {
-            if (r == null || isInterrupted()) {
-                return false;
-            }
-            migrating = (r instanceof MigrateTask);
             try {
+                if (r == null || isInterrupted()) {
+                    return false;
+                }
+
                 r.run();
             } catch (Throwable t) {
                 logger.warning(t);
+            } finally {
+                migrationQueue.afterTaskCompletion(r);
```

Comment:
Partition service already manages the whole migration process. Migration queue is only responsible for managing migration tasks. Therefore, having an extra callback method doesn't seem problematic to me. The other approach, wrapping the given runnable, is also ok but I don't think it is necessary and I want to keep our current approach.